### DISCUSSION TITLE
TypeError: Invalid URL

### DISCUSSION THREAD

**Discussion by RayJason** [remix-run/remix#9258] [Q&A]
I'm using react-three/drei for 3D. Put the model file in the `public`. And use the path: '/star. glb'

```ts
import { useGLTF } from '@react-three/drei'

// ...
const { nodes } = useGLTF('/star-icon-min.glb')
// ...
```

Although the model is loaded normally, there is an error:
<img width="570" alt="image" src="https://github.com/remix-run/remix/assets/15711610/e03b2a0f-3123-475c-9197-0e425446bda9">

I would like to know what is the best practice for static files in Remix.

**Comment by suhaotian** [comment]
Are you sure in `/public ` has this file `star-icon-min.glb`? I just create one file with same filename, then access `http://localhost:3000/star-icon-min.glb`, it's works

**Reply by RayJason** [reply]
Of course
```json
"@remix-run/node": "^2.8.1",
"@remix-run/react": "^2.8.1",
"@remix-run/serve": "^2.8.1",
```
![image](https://github.com/remix-run/remix/assets/15711610/d0128821-991e-4db0-a098-eee7683c2e32)

**Comment by suhaotian** [comment]
I check the source code, You can't pass path to useGLTF, you need pass url:

```tsx
const { nodes } = useGLTF('http://localhost:3000/star-icon-min.glb')
console.log(nodes)
```

That's why `Invalid URL`

**Reply by RayJason** [reply]
Meaning it's best to put the file in object storage ahead of time, using an online link?   
Is there a more elegant way to use local files? 🤣

**Reply by suhaotian** [reply]
Put it in yout public folder, then access the file via your server url + public path
